core/src/main/java/org/sonarsource/sonarlint/core/issue/IssueServiceImpl.java

```diff
@@ -175,28 +178,49 @@ public CompletableFuture<CheckStatusChangePermittedResponse> checkStatusChangePe
     }
     var issueKey = params.getIssueKey();
     var serverApi = serverApiOpt.get();
+
+    // when we checked against SonarCloud we don't have to check if it is SQ anymore
+    var isSonarCloudOrSQ104orNewer = serverApi.isSonarCloud();
+    isSonarCloudOrSQ104orNewer = isSonarCloudOrSQ104orNewer || is104orNewer(connectionId, serverApi);
```

Comment:
It looks a little bit strange to assign the variable twice, what about: 

> var isSonarCloudOrSQ104orNewer = serverApi.isSonarCloud() || is104orNewer(connectionId, serverApi);

It would read pretty well

Reply:
A leftover I missed